docs/proposals/multi-arch-image-mgmt.md

```diff
@@ -0,0 +1,328 @@
+  oras manifest index create --output index.json localhost:5000/hello linux-amd64 linux-arm64
+
+# Create an index and output the index to stdout, auto push will be disabled:
+  oras manifest index create localhost:5000/hello linux-arm64 --output - --pretty
+
+```
+
+### Update a multi-arch image
+
+```bash
+# Remove a manifest and add two manifests from an index tagged 'v1'. The tag will point to the updated index:
+  oras manifest index update localhost:5000/hello:v1 --add linux-amd64 --add linux-arm64 --remove sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9
+
+# Create a new index by updating an existing index specified by its digest:
+  oras manifest index update localhost:5000/hello@sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9 --add linux-amd64 --remove sha256:fd6ed2f36b5465244d5dc86cb4e7df0ab8a9d24adc57825099f522fe009a22bb
+
+# Merge manifests from the index 'v2-windows' to the index 'v2':
+  oras manifest index update localhost:5000/hello:v2 --merge v2-windows
+
+# Update an index and tag the updated index as 'v2.1.0' and 'v2':
+  oras manifest index update localhost:5000/hello@sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9 --add linux-amd64 --tag "v2.1.0" --tag "v2"
+
+# Update an index and save it locally to index.json, auto push will be disabled:
+  oras manifest index update --output index.json localhost:5000/hello:v2 --add v2-linux-amd64
+
+# Update an index and output the index to stdout, auto push will be disabled:
+  oras manifest index update --output - --pretty localhost:5000/hello:v2 --remove sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9
+```
+
+### Annotate a multi-arch image
+
+```bash
+# Add annotations to an image image via key and value
+oras manifest index update --oci-layout layout-dir:v1 --annotation “platform=multi-arch” 
+
+# Attach an annotation as a referrer to an OCI image index and its child image recursively
+oras attach --artifact-type application/vnd.rabbitnetworks.artifact.lifecycle --annotation "vnd.rabbitnetworks.artifact.lifecycle.end-of-life.date=2023-05-12" layout.tar:v1 --recursive
+```
+
+### Inspect a multi-arch image
+
+### List tag-based arch/platform information in a repo
+
+```
+oras repo tags --oci-layout layout-dir 
+```
+
+| Tag  | Digest  | Platform   |
+|--- | --- |  ---  |
+| v1 | sha:256@xxxx | multi-arch (linux/amd64, linux/arm64) |
+| v1-linux-amd64  |    sha:256@yyyy          |     linux/amd64   |
+| v1-linux-arm64  |    sha:256@zzzz          |     linux/arm64   |
+
+
+## Investigation on other client tools and industry
+
+Most of popular container client tools support create and push a multi-arch image using Docker manifest list or OCI image index format, but these tools **require users to push platform-specific image to the registry separately**. They don’t provide native support for local build experience.
```

Comment:
> They don’t provide native support for local build experience.

`docker buildx` or `buildkit` in generate does produce multi-arch image on build-time.